core/src/main/java/org/apache/iceberg/BaseTableScan.java

```diff
@@ -94,6 +94,13 @@ public TableScan useSnapshot(long scanSnapshotId) {
         tableOps(), table(), tableSchema(), context().useSnapshotId(scanSnapshotId));
   }
 
+  @Override
+  public TableScan useSnapshotRef(String snapshotRef) {
+    Preconditions.checkArgument(table().snapshot(snapshotRef) != null,
+        "Cannot find ref with name %s", snapshotRef);
+    return useSnapshot(table().snapshot(snapshotRef).snapshotId());
```

Comment:
The logic here needs to be a bit more complicated because you can time-travel within a branch:

```java
table.newScan().useRef("test-branch").asOfTime(t).planFiles()
```

`useSnapshot` is the most specific and can't be used with anything else. `useRef` with a tag name is the same way. `useRef` with a branch name is compatible with `asOfTime`, just like it is with `main` implicitly.